core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.js

```diff
@@ -49,7 +43,7 @@ module.exports = class puppeteerUtilities {
         this.page = await browser.newPage();
         await this.page.setViewport({ width: 0, height: 0 });
         // Accepting the alerts that appear in between the tests.
-        await this.page.on('dialog', async dialog => {
+        this.page.on('dialog', async(dialog) => {
```

Comment:
this is accepting the dialogs that appear in between the tests as we can handle alerts in puppeteer here. For example: when we publish a blog post and then move back to the /blog-dashboard from the /blog-dashboard#/blog_post_editor page, we need to accept that are you sure want to leave the page

Reply:
there are multiple scenarios possible in future also, like here in /admin page
<img width="1439" alt="Screenshot 2023-02-03 at 3 04 47 PM" src="https://user-images.githubusercontent.com/91013793/216565297-aed391e6-ae06-477c-95bb-adc7a0c7dba2.png">
and we always have to accept/confirm it, as such alerts will pop up due to our actions that we do in the tests and we won't do anything that we do not require.
@ashish-patwal can you please add a comment explaining the line in this PR itself

Reply:
@Shivkant-Chauhan @ashish-patwal That particular pop-up should be handled explicitly in the part of the tests that changes something on the admin dashboard. It shouldn't be handled generically because it's part of the specific action.

Reply:
@seanlip we can handle browser alerts only with such event listeners when the page instance got initialized, so no way to handle them in between the tests (where our action is popping up an alert). and it would be not good to have explicit if-else conditions here for handling such situations. I think it is good to handle such alerts generically and accept them all.

Reply:
Yeah, it seems like this is [unsolved](https://github.com/puppeteer/puppeteer/issues/1213) and all the docs I can find support what you're saying. 

I think it's still better not to do this generically but I don't think we have an alternative. Please add a comment in the code to explain why we are forced to do it this way.

Reply:
sure.. @ashish-patwal can you please add a comment as we discussed above for accepting the alerts generically.

Reply:
Sure !!

Reply:
@seanlip Can we actually draft another PR for this issue ( dialog popups ) ? Because this issue seems very unrelated to this PR. I can draft another PR for addressing this issue ( handling dialog popup in-between tests ).

Reply:
Yup, that's reasonable, I'll file an issue for that. Thanks for suggesting!

Reply:
Filed https://github.com/oppia/oppia/issues/17312